soh/src/code/z_room.c

```diff
@@ -256,89 +257,91 @@ s32 swapAndConvertJPEG(void* data) {
 
 void Room_DrawBackground2D(Gfx** gfxP, void* tex, void* tlut, u16 width, u16 height, u8 fmt, u8 siz, u16 tlutMode,
                            u16 tlutCount, f32 offsetX, f32 offsetY) {
-    Gfx* gfx = *gfxP;
-    uObjBg* bg;
-
-    bg = (uObjBg*)(gfx + 1);
-    gSPBranchList(gfx, (Gfx*)(bg + 1));
-
-    bg->b.imageX = 0;
-    bg->b.imageW = width * (1 << 2);
-    bg->b.frameX = offsetX * (1 << 2);
-    bg->b.imageY = 0;
-    bg->b.imageH = height * (1 << 2);
-    bg->b.frameY = offsetY * (1 << 2);
-    bg->b.imagePtr = tex;
-    bg->b.imageLoad = G_BGLT_LOADTILE;
-    bg->b.imageFmt = fmt;
-    bg->b.imageSiz = siz;
-    bg->b.imagePal = 0;
-    bg->b.imageFlip = CVarGetInteger(CVAR_ENHANCEMENT("MirroredWorld"), 0) ? G_BG_FLAG_FLIPS : 0;
-
-    // When an alt resource exists for the background, we need to unload the original asset
-    // to clear the cache so the alt asset will be loaded instead
-    // OTRTODO: If Alt loading over original cache is fixed, this line can most likely be removed
-    ResourceMgr_UnloadOriginalWhenAltExists((char*)tex);
-
-    if (ResourceMgr_ResourceIsBackground((char*)tex)) {
-        char* blob = (char*)ResourceGetDataByName((char*)tex);
-        swapAndConvertJPEG(blob);
-        bg->b.imagePtr = (uintptr_t)blob;
-    }
+    if (GameInteractor_Should(VB_DRAW_2D_BACKGROUND, true)) {
```

Comment:
since this is wrapping the entirety of `Room_DrawBackground2D`, i think it's worth considering other possibilities here.

i know for actors we do actionfunc replacement, but that won't really work here

if we want to stick with a `VB_SHOULD` here then i think just switching this to be

```suggestion
    if (!(GameInteractor_Should(VB_DRAW_2D_BACKGROUND, true))) {
        return;
    }
```

will clean it up a bit

Reply:
I can early return it instead, no problem. Wasnt really sure what the recommended approach was so i just surrounded all of it and returned it in my Register code to avoid adding too much

Reply:
i think this is missing a `!` now, the early return is in there but it's early returning when `DRAW_2D_BACKGROUND` is `true` instead of when it is `false`